src/JestConfigEditor.ts

```diff
@@ -34,4 +38,13 @@ export default class JestConfigEditor implements ConfigEditor {
 
     return configLoader;
   }
+
+  private overrideProperties(config: JestConfiguration) {
```

Comment:
You could use a `OVERRIDE_OPTIONS` object (`const` using `Object.freeze`) which you assign with `Object.assign`. 

At least add a comment for each option, so we later remember why we did it.